src/ui/win32/DialogBase.cpp

```diff
@@ -20,47 +22,58 @@ DialogBase::~DialogBase() noexcept
     }
 }
 
-static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+_Use_decl_annotations_
+_NODISCARD static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
-    DialogBase* pDialog = reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER));
+    const auto pDialog{ reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER)) };
+
+    // TBD: A dialog should not be calling DefWindowProc, it should return 0 
```

Comment:
This method applies to modal and non-modal windows. The particular line to which you seem to be commenting on is in case we get called after detaching the class instance pointer.